.buildkite/pipeline.yml

```diff
@@ -0,0 +1,39 @@
+common-params:
+  &docker-container
+  docker#v3.8.0:
+    image: "public.ecr.aws/automattic/android-build-image:v1.1.0"
+    propagate-environment: true
+    environment:
+      # DO NOT MANUALLY SET THESE VALUES!
+      # They are passed from the Buildkite agent to the Docker container
+      - "AWS_ACCESS_KEY"
+      - "AWS_SECRET_KEY"
+
+steps:
+  - label: "Lint & Checkstyle"
+    key: "lint_and_checkstyle"
+    plugins:
+      - *docker-container
+    command: |
+      ./gradlew lintRelease checkstyle
+    artifact_paths:
+      - "**/build/reports/lint-results**.*"
+      - "**/build/reports/checkstyle/checkstyle.*"
```

Comment:
@obsantos Do you mean making these steps a requirement for merging the app? That's typically what we do for our other apps. In this case, we'd actually want to make the publish to s3 step also a requirement. The only major disadvantage of these is the CI won't run for forked PRs, but there is a way around it and that's what we have been doing for other apps.

Otherwise, I am not sure I fully understand your question. To clarify, both `lint` & `checkstyle` tasks are run in this step, they are just combined for efficiency. I wondered if maybe that part wasn't obvious from the `./gradlew lintRelease checkstyle` line.

Also worth noting that I am still working on this PR, I just needed to open it so the artifacts would be published :)

Reply:
Making them a requirement (both merging and publishing) sounds great to me yeah.

My question/comment was related to removing or keeping `checkstyle` from the example app (you did remove it in https://github.com/tumblr/Backboard/commit/d524e1517f26257b81e4e1c977f0070ee6244caf). I was confused with this pipeline step and for some reason i thought it only applied to the `backboard` module, so i was suggesting adding another step for the example app module. Now I realized it applies to both 🤦 

Now that you removed `checkstyle` from the example app i wonder if this step will fail saying `checkstyle` task not found for `backboard-example`, if so maybe we need to change this to 
```
./gradlew lintRelease :backboard:checkstyle
```

Reply:
>Also worth noting that I am still working on this PR, I just needed to open it so the artifacts would be published :)

Haha ok got it, saw the PR and went for it 😅

Reply:
> Now that you removed checkstyle from the example app i wonder if this step will fail saying checkstyle task not found for backboard-example, if so maybe we need to change this to

The CI build was successful, so I think we are good. You can also verify that this is not an issue by running `./gradlew checkstyle --dry-run` command which instead of running the command will print out all the tasks that'll be run as part of it. The current result of this task is `:backboard:checkstyle` because `backboard` is the only module that has the `checkstyle` task.

I think for `checkstyle`, it makes more sense to go with the generic task because if we add more modules, it'd be covered. Having said that, I don't feel strongly about it, so if you prefer I can make the change.

Reply:
>I think for checkstyle, it makes more sense to go with the generic task because if we add more modules

good point 👍